packages/stellar-axelar-std-derive/src/contractimpl.rs

```diff
@@ -35,6 +44,58 @@ pub fn contractimpl(impl_block: &mut ItemImpl) -> Result<proc_macro2::TokenStrea
     })
 }
 
+/// Returns fn that finds all contract endpoints with an Env parameter
+fn all_contract_endpoints(
+    impl_block: &ItemImpl,
+) -> impl Fn(&mut ImplItem) -> Option<&mut ImplItemFn> {
+    let any_contract_endpoint = if impl_block.trait_.is_some() {
+        any
+    } else {
+        any_pub_fn
+    };
+
+    move |item| {
+        any_fn(item)
+            .and_then(any_contract_endpoint)
+            .filter(has_env_param)
+    }
+}
+
+fn instance_ttl_extension(method: &mut ImplItemFn) -> Result<(), syn::Error> {
+    let env_ident = parse_env_identifier(&method.sig.inputs)?;
+
```

Comment:
maybe add something like a `normalize_env_ref` helper, which normalizes to `&env`, and then it can be used anywhere without additional checks